### PR TITLE
Update grants to include revocation examples

### LINKED ISSUE #1955: `grants` config: removing all grants versus ignoring entirely
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/resource-configs/grants

### What part(s) of the page would you like to see updated?

Current expected behavior when:
- `grants` is empty / missing as a config
- how to remove all grants on a table that dbt previously applied (by adding an empty list of grantees)

### Additional information

https://github.com/dbt-labs/dbt-bigquery/issues/299

### PR DESCRIPTION
**Pull request by joellabes**
## What are you changing in this pull request and why?
Resolves #1955 and https://github.com/dbt-labs/docs.getdbt.com/issues/3259 - both issues related to grants so I figured I'd combine them to avoid merge conflicts. 

## Checklist
- [x] I have reviewed the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.

---
🚀 Deployment available! Here are the direct links to the updated files:

- https://docs-getdbt-com-git-joellabes-1955-remove-grants-dbt-labs.vercel.app/reference/resource-configs/grants

### CODE REVIEW

File: website/docs/reference/resource-configs/grants.md [hunk 1]
```diff
@@ -11,12 +11,12 @@ The grant resource configs enable you to apply permissions at build time to a sp
...
+You should define grants as resource configs whenever possible, but you might occasionally need to write grants statements manually and run them using [hooks](/docs/build/hooks-operations). For example, hooks may be appropriate if you want to:
 
-* Apply grants in a more complex or custom manner, beyond what the built-in grants capability can provide.
 * Apply grants on other database objects besides views and tables.
-* Take advantage of more-advanced permission capabilities offered by your data platform, for which dbt does not (yet!) offer out-of-the-box support using resource configuration.
 * Create more granular row- and column-level access, use masking policies, or apply future grants.
+* Take advantage of more-advanced permission capabilities offered by your data platform, for which dbt does not offer out-of-the-box support using resource configuration.
+* Apply grants in a more complex or custom manner, beyond what the built-in grants capability can provide.
```
Comment (joellabes) [inline website/docs/reference/resource-configs/grants.md]: This is not a change in content, just reordering to lead with the more specific examples and have the "etc" ones come second.

File: website/docs/reference/resource-configs/grants.md [hunk 2]
```diff
@@ -11,12 +11,12 @@ The grant resource configs enable you to apply permissions at build time to a sp
...
 
-dbt encourages you to use grants as resource configs whenever possible. In versions prior to Core v1.2, you were limited to using hooks for grants. Occasionally, you still might need to write grants statements manually and run them using hooks. For example, hooks may be appropriate if you want to:
+You should define grants as resource configs whenever possible, but you might occasionally need to write grants statements manually and run them using [hooks](/docs/build/hooks-operations). For example, hooks may be appropriate if you want to:
 
-* Apply grants in a more complex or custom manner, beyond what the built-in grants capability can provide.
 * Apply grants on other database objects besides views and tables.
-* Take advantage of more-advanced permission capabilities offered by your data platform, for which dbt does not (yet!) offer out-of-the-box support using resource configuration.
 * Create more granular row- and column-level access, use masking policies, or apply future grants.
+* Take advantage of more-advanced permission capabilities offered by your data platform, for which dbt does not offer out-of-the-box support using resource configuration.
```
Comment (matthewshaver) [inline website/docs/reference/resource-configs/grants.md]:
```suggestion
* Take advantage of more advanced permission capabilities offered by your data platform, for which dbt does not offer out-of-the-box support using resource configuration.
```

File: website/docs/reference/resource-configs/grants.md [hunk 3]
```diff
@@ -154,6 +154,83 @@ Now, the model will grant select to `user_a`, `user_b`, AND `user_c`!
...
+
+```yml
+models:
+  +grants:
+    select: ['user_a', 'user_b']
+```
+
+</File>
+
+If you delete the `+grants` section altogether, dbt will assume you no longer want it to manage grants, and will not change anything. To have dbt revoke all existing grants from a node, provide an empty list of grantees instead.
```
Comment (matthewshaver) [inline website/docs/reference/resource-configs/grants.md]:
```suggestion
If you delete the entire `+grants` section, dbt assumes you no longer want it to manage grants and doesn't change anything. To have dbt revoke all existing grants from a node, provide an empty list of grantees.
```

File: website/docs/reference/resource-configs/grants.md [hunk 4]
```diff
@@ -154,6 +154,83 @@ Now, the model will grant select to `user_a`, `user_b`, AND `user_c`!
...
+models:
+  +grants:
+    select: "{{ ['user_a', 'user_b'] if target.name == 'prod' else ['user_c'] }}"
+```
+
+</File>
+
+## Revoking grants
+
+dbt will only modify grants on a node (including revocation) when a `grants` configuration is attached to that node. For example, imagine you had originally specified the following grants in `dbt_project.yml`:
```
Comment (matthewshaver) [inline website/docs/reference/resource-configs/grants.md]:
```suggestion
dbt only modifies grants on a node (including revocation) when a `grants` configuration is attached to that node. For example, imagine you had originally specified the following grants in `dbt_project.yml`:
```

**Comment by matthewshaver** [comment]
Thanks @joellabes Looks great! We appreciate the help. I made some quick editorial changes. I removed some future tense to lure users into the now.

**Review by matthewshaver — APPROVED** [review]

**Pull request merged**